syncapi/streams/stream_invite.go

```diff
@@ -56,6 +59,16 @@ func (p *InviteStreamProvider) IncrementalSync(
 	for roomID := range retiredInvites {
 		if _, ok := req.Response.Rooms.Join[roomID]; !ok {
 			lr := types.NewLeaveResponse()
+			lr.Timeline.Events = append(lr.Timeline.Events, gomatrixserverlib.ClientEvent{
+				// fake event ID which muxes in the to position
+				EventID:        "$stub-retired-invite-" + roomID + "-" + strconv.FormatInt(int64(to), 10),
```

Comment:
We should probably strip `roomID` so that it only contains the alphanumeric characters and not `!`, `:` etc. I don't think `!` is a valid character for the event ID in any room version and `:` has a special meaning as a separator in room version 1. Otherwise we run the risk of sending something that will confuse clients.